### ISSUE TITLE
Model nested resources without delete operations as properties of the parent

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi-azure-native#2452]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Related to https://github.com/pulumi/pulumi-azure-native/issues/1112

There's some resources in the Azure API specs which only provide GET and PUT operations - no DELETE. By default we assume that the "delete" operation should be implemented by doing a PUT of an empty object. If an empty object is not appropriate, then we can set an alternative default object in in `defaultResourcesState.go`.

In practice, most of these resource paths are below another resource, and get deleted along with the parent. Therefore these should be modelled inside of the parent resource - adding their args alongside the parents'. When that property then changes, we would perform an additional PUT to the nested endpoint to set its properties.
